jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/CredentialWriter.kt

```diff
@@ -55,6 +58,15 @@ class CreateOrUpdateCredentialProfilesAction @TestOnly constructor(
 
         localFileSystem.refreshFiles(virtualFiles, false, false) {
             virtualFiles.forEach {
+                if (it.fileType == FileTypes.UNKNOWN) {
+                    ApplicationManager.getApplication().runWriteAction {
+                        FileTypeManagerEx.getInstanceEx().associatePattern(
```

Comment:
Does this register this pattern application wide?

Reply:
It seems it does.... So we either do this or skip empty files or crash....

Ideas?

Reply:
It seems so. https://github.com/JetBrains/intellij-community/blob/master/platform/platform-api/src/com/intellij/openapi/fileTypes/FileTypeManager.java#L51-L58

Reply:
Long term it may make sense to make a custom config/credential language so we can provide docs for properties in the IDE